### DISCUSSION TITLE
Mission bug report:  Ablub's Invention - Reclaim the Probe

### DISCUSSION THREAD

**Discussion by ChuckStarchaser** [endless-sky/endless-sky#7523] [General]
This is a double-bug report.

I took the mission from this Arachnid lady at Ablub's Invention to go searching for the lost probe,
which it turns out was bought by a pirate, who sold it to a junk yard, from where it was bought
by another pirate, but not from the South, so I went towards the pirate base in Glenah, and found the three Quicksilver pirate ships I was looking for.  The first one I disabled and boarded had the probe, so I got the message that I had the probe in my cargo, and all I had to do was bring it back to Ablub's Invention.  So here's how the first bug can be engaged:

Disable another of the Quicksilver ships, and search it.  A popup says "nothing here to see;  maybe look in another ship".
I was looking at another ship out of curiosity;  I already had the probe.  What I did not forsee is the fact that looking at another ship somehow made the probe disappear from my cargo!!!
Well, at least that's my theory of how it happened.
What I do know for sure is that the next time I landed there was nothing in my cargo, and when I looked at the Missions it told me I was still expected to find the ships and take the probe...  Doh!  But here comes the second bug:

Going back to the system those ships don't re-spawn.  So the mission code acknowledged that the ships had been raided, but disacknowledged my possession of the probe;  so now the mission is completely lost for me, as I landed several times, so the autosave from before the mission is already gone.

**Comment by warp-core** [comment]
@ChuckStarchaser for future reference, bug reports should be issues.

@Arachi-Lover this mission is one of yours, I think?

**Reply by Arachi-Lover** [reply]
> @Arachi-Lover this mission is one of yours, I think?

Yes.

So, the missions in question function this way:
There are 3 Quicksilvers, 1 spawned by the main mission (the one that has the probe), and 2 decoy ones, spawned by a separate mission each.
The destination for this mission is the planet in Coalition space you get the initial mission prompt for this mission chain, but the mission indicator there will be greyed out until you recover the probe.
The recovery, upon boarding the correct Quicksilver, is done by a condition added by an apply-set:
```
npc board
		personality staying nemesis target
		government "Bounty"
		system "Persian"
		ship "Marauder Quicksilver (Engines) (Safe)" "Cohen"
		conversation
			apply
				set "idriss probe recovered"
```
The bit about getting the probe into your cargo hold is just some text fluff to say you've got it back, safe and sound.
You don't actually get cargo added to your ship, so it wouldn't show up in the cargo list menu anyhow. Boarding the other Quicksilvers after the fact won't remove the condition, so it shouldn't lock you out of the mission.
Getting the condition should make it so that the mission marker to Ablub isn't greyed out anymore, so one should be able of completing the mission (which is the only thing that clears the `"idriss probe recovered"` condition).

Can you really not complete the mission anymore? I didn't run into this issue when testing, and from just looking at the mission code now don't see how that'd have happened.

**Comment by ChuckStarchaser** [comment]
Okay, quick answer:  I don't know;  I haven't tried going back to Ablub's Invention.  I'm going to try, in one minute;  I just wanted to defend my report first by saying that the status of the mission would normally change to something like "Bring back the probe", as opposed to continuing to say "Reclaim the probe";  or continue to call it the same, but the description should change to "bring it back" rather than "find the fighter that has it".  It is this legend, in addition to the absence of anything in the cargo, that convinced me the mission was lost.  I'm going to try now ...

**Comment by ChuckStarchaser** [comment]
Yes, you're right;  the mission does complete.  Thanks.  Still, I think the legend needs to change.

Well, perhaps there should also be a way to add a text-item to the hold via the mission scripting.
I'll put this into Ideas.

**Comment by Arachi-Lover** [comment]
Yeah, I see how it could get confusing with the legend.
With current mechanics though I don't think this could be made to work the right way to eliminate the possible confusion, as I don't think we can make a specific NPC board action remove a mission from the active missions list, and add in a new mission immediately after said boarding to update the mission actions (or alternatively edit/update the description of an existing active mission).

If we can do that though and it's just the case that I don't know how to implement that, someone could direct me how to do the fix, or do it themselves if they prefer.
